### PR TITLE
MCI Refresh

### PR DESCRIPTION
**Pull request by warrenbailey**
This PR allows us to process the new MCI refreshed schema from EQ.

### CODE REVIEW

File: transform/transformers/PCKTransformer.py [hunk 1]
```diff
@@ -17,7 +19,11 @@ class PCKTransformer(object):
         },
         "139": {
             "0001": "Q01B",
+        },
+        "mci": {
+            "refresh": "RSI10B", # TODO remove this once EQ ready for switch
```
Comment (necrophonic) [inline transform/transformers/PCKTransformer.py]: May be worth raising a card in the backlog to cover this for when it comes it to keep visibility.

File: transform/views/main.py [hunk 2]
```diff
@@ -52,11 +52,13 @@ def server_error(error=None):
 def get_survey(survey_response):
     try:
         form_id = survey_response['collection']['instrument_id']
+        logger.info("Loading survey {0}-{1}.json".format(survey_response['survey_id'], form_id))
```
Comment (necrophonic) [inline transform/views/main.py]: Are we logging the tx_id with the log statements?

File: transform/views/main.py [hunk 3]
```diff
@@ -195,6 +200,7 @@ def common_software(sequence_no=1000, batch_number=0):
             except IOError as e:
                 return client_error("CS:Could not create zip buffer: {0}".format(repr(e)))
     except Exception as e:
+        logger.exception("commono software method error")
```
Comment (necrophonic) [inline transform/views/main.py]: Typpo

File: transform/views/main.py [hunk 4]
```diff
@@ -128,6 +130,9 @@ def render_pdf():
 
     except IOError as e:
         return client_error("PDF:Could not render pdf buffer: {0}".format(repr(e)))
+    except Exception as e:
+        logger.exception("PDF generation failed")
```
Comment (necrophonic) [inline transform/views/main.py]: Can we add any info?

File: transform/views/main.py (hunk 2 above)
Comment (warrenbailey) [inline transform/views/main.py]: Done

File: transform/views/main.py (hunk 4 above)
Comment (warrenbailey) [inline transform/views/main.py]: Done

File: transform/views/main.py [hunk 5]
```diff
@@ -52,11 +52,15 @@ def server_error(error=None):
 def get_survey(survey_response):
     try:
         form_id = survey_response['collection']['instrument_id']
+        survey_id = survey_response.get("survey_id", "N/A")
+        tx_id = survey_response.get("tx_id", "N/A")
+        logger.info("Loading survey {0}-{1}.json for tx_id {2}".format(survey_id, form_id, tx_id))
```
Comment (eldeal) [inline transform/views/main.py]: Can survey_id, form_id, tx_id and filename all be KV pairs please

File: transform/views/main.py [hunk 6]
```diff
@@ -52,11 +52,15 @@ def server_error(error=None):
...
 
         fp = os.path.join(
             ".", "transform", "surveys",
             "{0}.{1}.json".format(survey_response['survey_id'], form_id)
         )
+        logger.info("Opening file {0} for tx_id {1}".format(fp, tx_id))
```
Comment (eldeal) [inline transform/views/main.py]: filepath and tx_id as KV pairs please

File: transform/views/main.py [hunk 7]
```diff
@@ -128,6 +132,11 @@ def render_pdf():
 
     except IOError as e:
         return client_error("PDF:Could not render pdf buffer: {0}".format(repr(e)))
+    except Exception as e:
+        survey_id = survey_response.get("survey_id", "N/A")
+        tx_id = survey_response.get("tx_id", "N/A")
+        logger.exception("PDF:Generation failed for survey with id {0} and tx_id {1)", survey_id, tx_id)
```
Comment (eldeal) [inline transform/views/main.py]: survey_id and tx_id as KV pairs please - error message can just be "PDF generation failed" or something :+1:

File: transform/views/main.py [hunk 8]
```diff
@@ -195,6 +204,8 @@ def common_software(sequence_no=1000, batch_number=0):
             except IOError as e:
                 return client_error("CS:Could not create zip buffer: {0}".format(repr(e)))
     except Exception as e:
+        tx_id = survey_response.get("tx_id", "N/A")
+        logger.exception("CS:could not create files for survey with id {0} and tx_id {1}".format(survey_id, tx_id))
```
Comment (eldeal) [inline transform/views/main.py]: KV pairs please!

File: transform/views/main.py (hunk 5 above)
Comment (warrenbailey) [inline transform/views/main.py]: Yep i'll change

File: transform/views/main.py (hunk 7 above)
Comment (warrenbailey) [inline transform/views/main.py]: ok

File: transform/views/main.py (hunk 8 above)
Comment (warrenbailey) [inline transform/views/main.py]: You sure you want KV pairs now ;)

File: transform/views/main.py [hunk 9]
```diff
@@ -52,14 +52,19 @@ def server_error(error=None):
...
 
         fp = os.path.join(
             ".", "transform", "surveys",
             "{0}.{1}.json".format(survey_response['survey_id'], form_id)
         )
+        logger.info("Opening file", file=fp, tx_id=tx_id)
         with open(fp, 'r') as json_file:
             return json.load(json_file)
     except IOError:
+        logger.exception("Error opening file")
```
Comment (JamesGardiner) [inline transform/views/main.py]: Add the file path to the log message

**Comment by eldeal** [comment]
Update CHANGELOG please :)

**Pull request merged**